Add Sysbox as an option to run Kubernetes inside unprivileged containers.

ctalledo opened the pull request:
Sysbox is an open-source container runtime (similar to "runc") that supports
running VM-workloads such as Docker and Kubernetes inside unprivileged containers or pods.

Sysbox containers always use the Linux user-namespace for isolation, plus
specially crafted proc and sys filesystems, some syscall interception,
filesystem ID-mapping, and more.

It's possible to run Kubernetes, K3s, K0s, inside containers or pods deployed
with Sysbox. This commit aims to make the Kubernetes community aware of this
option.

Signed-off-by: Cesar Talledo <ctalledo@nestybox.com>

ctalledo commented:
/check-cla

ctalledo commented:
/check-cla

ctalledo commented:
Not sure why the "cla/linuxfoundation" check failed, as I've signed the CLA and the EasyCLA checker passed.

Arhell reviewed (commented):
/check-cla

sftim commented on content/en/docs/tasks/administer-cluster/kubelet-in-userns.md:
```diff
@@ -49,6 +49,21 @@ Rootless Podman is not supported.
 
 <!-- Supporting rootless podman is discussed in https://github.com/kubernetes/minikube/issues/8719 -->
 
+## Running Kubernetes inside Unprivileged Containers
+
+### sysbox
+
+[Sysbox](https://github.com/nestybox/sysbox) is an open-source container runtime
+(similar to "runc") that supports running system-level workloads such as Docker
+and Kubernetes inside containers isolated via the Linux user namespace.
+
+See Kubernetes-in-Docker in the [Sysbox docs][https://github.com/nestybox/sysbox/blob/master/docs/quickstart/kind.md] for more info.
+
+Note that Sysbox supports running Kubernetes inside unprivileged containers
+without requiring Cgroup v2 and without the `KubeletInUserNamespace` feature
+gate. It does this by exposing specially crafted proc and sys filesystems inside
+the container plus several other advanced techniques.
```
Reviewers: take care to check whether this change fits the [content guide](https://kubernetes.io/docs/contribute/style/content-guide/)

divya-mohan0209 replied:
@ctalledo : Per the content guide linked by @sftim above, this is third party content and therefore, will require a third party content tag to be added. 

{{% thirdparty-content %}}

divya-mohan0209 commented:
@ctalledo : Please sing the CLA

divya-mohan0209 reviewed (changes requested):
Please sign the CLA & address the review.

ctalledo commented:
/check-cla

ctalledo commented:
Hi @divya-mohan0209, thanks for the review.

I've added the `{{% thirdparty-content %}}` tag to the section I added.

With regards to the CLA, I"ve signed it a few days ago, not sure with the `cla/linuxfoundation` checker is not picking this up, but as you can see in the 3rd comment of this issue, the EasyCLA checker did pass.

I've tried retriggering the `cla/linuxfoundation` checker by creating a comment with `/check-cla`, but it does not seem to work.

Thanks again!

sftim commented:
We're switching between two bots for the CLA signing

https://github.com/kubernetes/website/pull/31330#issuecomment-1012495785 is the instructions that you need to follow right now.

ctalledo commented:
Hi @sftim, 

> [#31330 (comment)](https://github.com/kubernetes/website/pull/31330#issuecomment-1012495785) is the instructions that you need to follow right now.

Yes, I've done this already (few days ago) but the `cla/linuxfoundation` checker does not seem to pick it up. 

I just clicked on the link again to double-check, and takes me to a Linux Foundation website that confirms that I am authorized to contribute code through my membership with "CNCF - Nestybox, Inc." (Nestybox is the company I work for). So I am not sure what am I doing wrong here. Sorry for the trouble.

jimangel commented:
Hey there! @ctalledo, looks like you haven't signed the CLA yet. Could I please have you do that? https://github.com/kubernetes/community/blob/master/CLA.md

ctalledo commented:
> Hey there! @ctalledo, looks like you haven't signed the CLA yet. Could I please have you do that? https://github.com/kubernetes/community/blob/master/CLA.md

Hi @jimangel, I did sign the CLA (see [this comment](https://github.com/kubernetes/website/pull/31330#issuecomment-1012495786)) but for some reason cla/linuxfoundation checker is not picking it up.

I tried re-triggering it with `/check-cla` but it did not help :(

Babapool commented:
Hi @ctalledo Thank you for your contribution. 
I might have a probable solution to this  problem (though not sure of it and there is no harm in trying it): Please try posting the comment "I signed it" to the PR—this is needed for the system to recheck your contribution authorization. Thanks.

sftim commented:
If you don't mind trying that @ctalledo, you could try opening a brand new PR with the same commits (maybe on a new branch)?

ctalledo commented:
I signed it

ctalledo commented:
Hi @Babapool,

> Please try posting the comment "I signed it" to the PR to recheck your contribution authorization. Thanks.

I did this but it does not appear to help.

I'll follow @sftim advice and re-open a fresh new PR, with the hope that this time the CLA checker will pick it up.

Closing this PR now, will open a fresh new one soon.

Thanks for the help.
